sklearn/ensemble/gradient_boosting.py

```diff
@@ -1184,6 +1184,37 @@ def _validate_y(self, y):
         # Default implementation
         return y
 
+    def apply(self, X):
+        """Apply trees in the ensemble to X, return leaf indices.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape = [n_samples, n_features]
+            The input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csr_matrix``.
+
+        Returns
+        -------
+        X_leaves : array_like, shape = [n_samples, n_estimators, n_classes]
```

Comment:
Unfortunately this docstring will also be use for the `GradientBoostingRegressor` model where the concept of class is meaningless. I think the shape will be `[n_samples, n_estimators, 1]` or `[n_samples, n_estimators]` in that case.